src/Transformer.js

```diff
@@ -0,0 +1,39 @@
+export default class Transformer {
+  /**
+   * List of rules to transform.
+   *
+   * @type {{}}
+   */
+  rules = {}
+
+  /**
+   * Transformer constructor.
+   *
+   * @param {object} dependencies List of dependencies.
+   */
+  constructor (dependencies = {}) {
+    for (let key in dependencies) {
+      this[key] = dependencies[key]
```

Comment:
This will make it impossible for a factory to understand what to give to the transformer.